nanoqm/workflows/workflow_coop.py

```diff
@@ -102,8 +115,17 @@ def compute_overlap_and_atomic_orbitals(
     element_1 = config["coop_elements"][0]
     element_2 = config["coop_elements"][1]
 
-    element_1_index = [i for i, s in enumerate(mol) if element_1.lower() in s]
-    element_2_index = [i for i, s in enumerate(mol) if element_2.lower() in s]
+    if config["elements_coordination"] is None:
+        element_1_index = [i for i, s in enumerate(mol) if element_1.lower() in s]
+        element_2_index = [i for i, s in enumerate(mol) if element_2.lower() in s]
+
+    # For the two selected elements, only the indices corresponding to a given coordination number
+    else:
```

Comment:
Fyi, you can also make `nano-CAT` an optional dependency and then raise any potential `ImportError`s whenever a code branch is reached that _does_ require `nano-CAT`:


``` python
from typing import Optional
try:
    from nanoCAT import coordination_number
    NANOCAT_EX: Optional[ImportError] = None
except ImportError as ex:
    NANOCAT_EX = ex


def compute_overlap_and_atomic_orbitals(...):
    if config["elements_coordination"] is None:
        ...
    elif NANOCAT_EX is not None:
        raise NANOCAT_EX
    else:
        coord = coordination_number(geometry)
        ...
```